src/phew/server.py

```diff
@@ -116,73 +58,21 @@ def __str__(self):
 body: {self.body}"""
 
 
-content_type_map = {
-    "html": "text/html",
-    "jpg": "image/jpeg",
-    "jpeg": "image/jpeg",
-    "svg": "image/svg+xml",
-    "json": "application/json",
-    "png": "image/png",
-    "css": "text/css",
-    "js": "text/javascript",
-    "csv": "text/csv",
-}
-
-
-class FileResponse(Response):
-    def __init__(self, file, status=200, headers={}):
-        self.status = 404
-        self.headers = headers
-        self.file = file
-
-        try:
-            if (os.stat(self.file)[0] & 0x4000) == 0:
-                self.status = 200
-
-                # auto set content type
-                extension = self.file.split(".")[-1].lower()
-                if extension in content_type_map:
-                    headers["Content-Type"] = content_type_map[extension]
-
-            headers["Content-Length"] = os.stat(self.file)[6]
-        except OSError:
-            return False
-
-
 class Route:
-    def __init__(self, path, handler, methods=["GET"]):
+    def __init__(self, path, handler, method="GET"):
         self.path = path
-        self.methods = methods
+        self.method = method
         self.handler = handler
-        self.path_parts = path.split("/")
 
     # returns True if the supplied request matches this route
     def matches(self, request):
-        if request.method not in self.methods:
+        if request.method != self.method:
             return False
-        compare_parts = request.path.split("/")
-        if len(compare_parts) != len(self.path_parts):
-            return False
-        for part, compare in zip(self.path_parts, compare_parts):
-            if not part.startswith("<") and part != compare:
-                return False
-        return True
+        return request.path == self.path
```

Comment:
removed url query parameter support so now all we need to do to is compare the path directly.